### ISSUE TITLE
ready: schema mismatch and hang prevent canonical task-queue enumeration

### ISSUE THREAD

**Issue by verioussmith** [Dicklesworthstone/beads_rust#483]
## Outcome blocked

An agent cannot determine which project-local tasks are ready, so it cannot safely continue an authorized work queue or prove that the queue is exhausted. The failure must not be interpreted as “no ready work.”

## Environment

- `br` / `bd`: **0.2.16**
- Repository-local `.beads/` store
- JSONL scale: **3,031 records**

No project records are attached to this report.

## Reproduction 1: database-backed `ready`

```bash
cd <project>
bd ready --json
```

Observed:

```text
Warning: Daemon took too long to start (>5s). Running in direct mode.
Error: failed to get ready work: sqlite3: SQL logic error: no such column: i.crystallizes
```

## Reproduction 2: JSONL-only fallback

```bash
timeout 30 br ready --no-db --json
```

Observed: exits `124` after 30 seconds with no stdout or stderr.

A later `br doctor --json` completed and reported:

- database count `0`, JSONL count `3031`;
- JSONL newer than database / import pending;
- preserved database recovery artifacts;
- eight fully unblocked open issues that `br ready` should surface.

A normal `bd ready --json` also hung beyond 20 seconds with no output after the original schema error.

## Expected behavior

`ready` must either return the deterministic ready set or fail quickly with a typed, actionable, non-destructive schema/migration error. JSONL-only mode must provide a bounded read path rather than hanging.

## Acceptance

1. Old/current database fixtures migrate or query compatibly when `crystallizes` is absent.
2. `ready` has bounded startup/query behavior and never hangs silently.
3. `ready --no-db --json` returns a deterministic result for a 3k-record dependency graph within a documented budget.
4. A read operation cannot leave a zero-row database or unexplained recovery artifacts; interrupted import/recovery rolls back safely.
5. Doctor reports the exact incompatible schema/generation and one safe recovery command.
6. Regression covers daemon timeout → direct mode, absent-column database, JSONL-newer state, and JSONL-only scale.

**Comment by Dicklesworthstone** [comment]
Verified each part against the current code; two of the three symptoms are not `br`, the third was real and is improved in 2f9518a2 (with the bulk of the code landing in 056d271d). Ships in the next release.

**1. `no such column: i.crystallizes` and the daemon warning.** Both messages come from the Go `bd` binary (`Daemon took too long to start`, `sqlite3: SQL logic error`). `br` has no daemon, never references a `crystallizes` column in SQL, and its schema does not define one. What you hit is a newer Go `bd` querying a database written by `br` (or an older `bd`), whose schema lacks the HOP columns that Go build expects. `br` and `bd` share the JSONL format, not a live database: keep one tool per `.beads/beads.db`, and if you are switching to `br`, let it rebuild the database from `issues.jsonl` (`br sync --import-only --rebuild`, or simply delete the `bd`-written `beads.db` and run any `br` command). The "database count 0 / JSONL 3031 / recovery artifacts" doctor report is that situation seen from `br`'s side: it found a database it could not use, preserved it, and had not yet imported the JSONL.

**2. `br ready --no-db --json` "hang".** Reproduced with a generated 3,031-record graph (open/closed mix, up to three `blocks` edges each): it does not hang, but on 0.5.x it took ~11.6 s, which is over your 30 s budget on slower hardware and certainly on the 0.2.x line. Timing breakdown (`-vv` now logs every import phase; numbers from the new scale probe on a loaded CI box, so relative rather than absolute): parse and validate 22 ms; issue rows, relations, and export hashes 2.6 s (one prepared statement per row in the private FrankenSQLite database); orphan cleanup 0.2 s; blocked-cache rebuild 1.3 s; semantic re-read 0.5 s; foreign-key verification 1.7 s; the ready query itself 0.35 s. `--no-db` rebuilds that private database from the whole JSONL on every invocation; the same import runs once in the default mode, after which `br ready --json` answers in 0.16 s on this dataset.

What changed:
- The import proves the relation tables empty once per transaction and then takes the insert-only relation path for every record, instead of probing each record's labels/dependencies/comments first. Previously that path was reserved for reviewed fresh-replacement imports; it now also covers every `--no-db` open and every first import.
- Import phases log elapsed times at `-vv`, and the `--no-db` rebuild shows a progress bar on an interactive stderr so the wait is never silent (`--quiet` and redirected stderr stay silent).
- `docs/CLI_REFERENCE.md` documents the `--no-db` cost model and recommends the database-backed mode for large workspaces.
- Tests: insert-only path plus re-import into a populated database (relations replaced, not duplicated), and an explicit 3,031-record scale probe (`cargo test --lib no_db_scale_import -- --ignored --nocapture`).

The remaining per-statement cost is in the engine (each row is still one prepared statement round trip); a batched multi-row insert for issues is the next lever and is tracked separately. For a 3k-record queue the practical advice stands: use the default mode, where `ready` is a sub-second SQLite query after a one-time import.

On your acceptance list: (1) is a `bd`/`br` cross-tool mismatch, not a migration `br` can perform; (2)/(3) addressed as above; (4) a read never touches the on-disk database in `--no-db` mode, and the recovery artifacts you saw are `br` preserving a foreign database rather than deleting it; (5) `br doctor` already names the incompatible schema and `br sync --import-only --rebuild` as the recovery. Bead: beads_rust-x0q4.
